src/components/TransactionFlow/TxReceiptTotals.tsx

```diff
@@ -0,0 +1,127 @@
+  const feeWei = totalTxFeeToWei(gasPrice, gasUsed);
+
+  const feeFormatted = parseFloat(fromWei(feeWei, 'ether')).toFixed(6);
+
+  const valueWei = Wei(value);
+
+  const totalWei = feeWei.add(valueWei);
+
+  const totalEtherFormatted = parseFloat(fromWei(totalWei, 'ether')).toFixed(6);
+
+  return (
+    <>
+      <div className="TransactionReceipt-row">
+        <div className="TransactionReceipt-row-column">
+          <SIcon type="tx-send" alt="Sent" />
+          {translate('CONFIRM_TX_SENT')}
+        </div>
+        <div className="TransactionReceipt-row-column rightAligned">
+          <AssetIcon uuid={asset.uuid} size={'24px'} />
+          <Amount
+            assetValue={`${parseFloat(assetAmount).toFixed(6)} ${asset.ticker}`}
+            fiat={{
+              symbol: getFiat(settings).symbol,
+              ticker: getFiat(settings).ticker,
+              amount: convertToFiat(parseFloat(assetAmount), assetRate).toFixed(2)
+            }}
+          />
+        </div>
+      </div>
+
+      <div className="TransactionReceipt-row">
+        <div className="TransactionReceipt-row-column">
+          <SIcon type="tx-fee" alt="Fee" /> {translate('CONFIRM_TX_FEE')}
+        </div>
+        <div className="TransactionReceipt-row-column rightAligned">
+          <AssetIcon uuid={asset.uuid} size={'24px'} />
+          <Amount
+            assetValue={`${feeFormatted} ${baseAsset.ticker}`}
+            fiat={{
+              symbol: getFiat(settings).symbol,
+              ticker: getFiat(settings).ticker,
+              amount: convertToFiat(parseFloat(feeFormatted), baseAssetRate).toFixed(2)
+            }}
+          />
+        </div>
+      </div>
+
+      <div className="TransactionReceipt-divider" />
+
+      <div className="TransactionReceipt-row">
+        <div className="TransactionReceipt-row-column">
+          <SIcon type="tx-sent" alt="Sent" />
+          {translate('TOTAL')}
+        </div>
+        <div className="TransactionReceipt-row-column rightAligned">
+          <AssetIcon uuid={asset.uuid} size={'24px'} />
+          {asset.type === 'base' ? (
+            <Amount
+              assetValue={`${totalEtherFormatted} ${asset.ticker}`}
+              fiat={{
+                symbol: getFiat(settings).symbol,
+                ticker: getFiat(settings).ticker,
+                amount: convertToFiat(parseFloat(totalEtherFormatted), assetRate).toFixed(2)
+              }}
+            />
+          ) : (
+            <Amount
+              assetValue={`${assetAmount} ${asset.ticker}`}
+              bold={true}
+              baseAssetValue={`+ ${totalEtherFormatted} ${baseAsset.ticker}`}
+              fiat={{
+                symbol: getFiat(settings).symbol,
+                ticker: getFiat(settings).ticker,
+                amount: (
+                  convertToFiat(parseFloat(assetAmount), assetRate) +
+                  convertToFiat(parseFloat(totalEtherFormatted), baseAssetRate)
+                ).toFixed(2)
```

Comment:
I would also use BN for any amount maths

Reply:
BN math shouldn't be required for adding fiat amounts should it?

Reply:
afaik any maths with floats should be avoided, it could introduce inaccuracy.